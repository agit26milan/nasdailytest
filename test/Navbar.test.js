import React from 'react'
import { shallow, configure } from 'enzyme'
import NavbarComponent from '../src/Components/Navbar'
import { Navbar } from 'react-bootstrap'
import Adapter from 'enzyme-adapter-react-16'
import TestRenderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('Should make snapshot', () => {
  const wrapper = TestRenderer.create(<NavbarComponent />)
  it('should same as snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Render navbar component', () => {
  const wrapper = shallow(<NavbarComponent />)
  it('Should have navbar componet', () => {
    expect(wrapper.find(Navbar)).toHaveLength(1)
    expect(wrapper.find(Navbar.Brand)).toHaveLength(1)
    expect(wrapper.find(Navbar.Toggle)).toHaveLength(1)
  })
})
