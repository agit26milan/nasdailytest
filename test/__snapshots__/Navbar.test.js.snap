// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should make snapshot should same as snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
  id="navbar"
>
  <a
    className="hacker navbar-brand"
    href="/"
  >
    HACKER
    <b
      className="news"
    >
      NEWS
    </b>
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
</nav>
`;
